stake-pool/js/src/test.ts

```diff
@@ -0,0 +1,194 @@
+            'hex',
+          ),
+        }),
+        reserveStake: new schema.PublicKey({
+          value: new BN(
+            '74a5b1ab8442103baa8bd39ab8494eb034e96035ac664e1693bb3eef458761ee',
+            'hex',
+          ),
+        }),
+        poolMint: new schema.PublicKey({
+          value: new BN(
+            '8722bf107b95d2620008d256b18c13fa3a46ab7f643c24cf7656f57267563e00',
+            'hex',
+          ),
+        }),
+        managerFeeAccount: new schema.PublicKey({
+          value: new BN(
+            new Buffer(
+              'b783b4dcd341cbca22e781bbd49b2d16908a844a21b98e26b69d44fc50e1db0f',
+              'hex',
+            ),
+          ),
+        }),
+        tokenProgramId: new schema.PublicKey({
+          value: new BN(
+            'a900ff7e85f58c3a91375b5fed85b41cac79ebce46e1cbd993a165d7e1f6dd06',
+            'hex',
+          ),
+        }),
+        totalStakeLamports: new BN('0', 'hex'),
+        poolTokenSupply: new BN('0', 'hex'),
+        lastUpdateEpoch: new BN('7c', 'hex'),
+        fee: new schema.Fee({
+          denominator: new BN('3e8', 'hex'),
+          numerator: new BN('38', 'hex'),
+        }),
+      });
+
+      for (const key in decodedData) {
+        if (expectedData[key] instanceof BN) {
+          assert.ok(expectedData[key].eq(decodedData[key]));
+        } else {
+          for (const subkey in decodedData[key]) {
+            if (decodedData[key][subkey] instanceof BN) {
+              assert.ok(decodedData[key][subkey].eq(expectedData[key][subkey]));
+            } else {
+              assert.deepStrictEqual(
+                expectedData[key][subkey],
+                decodedData[key][subkey],
+              );
+            }
+          }
+        }
+      }
+    });
+  });
+
+  describe('ValidatorListAccount', () => {
+    it('should successfully decode ValidatorListAccount account data', () => {
+      const decodedData = schema.ValidatorList.decodeUnchecked(
+        new Buffer(
+          '020a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
```

Comment:
We should include some data in this to make it test the full deserialization behavior

Reply:
Done in https://github.com/solana-labs/solana-program-library/pull/1656/commits/01f557a157dcf11837dc097f21df1ce4bd42c23c